File: jobboard-backend/src/main/java/com/amar/jobboard/service/JobService.java
package com.amar.jobboard.service;

import org.springframework.data.domain.Page;

import com.amar.jobboard.dto.JobRequest;
import com.amar.jobboard.dto.JobResponse;

public interface JobService {

    JobResponse createJob(JobRequest request, String recruiterEmail);

    JobResponse updateJob(Long id, JobRequest request);

    void deleteJob(Long id);

    Page<JobResponse> getAllJobs(int page, int size);

    JobResponse getJobById(Long id);

    Page<JobResponse> searchJobs(String keyword, int page, int size);
}

File: jobboard-frontend/src/services/jobService.js
import axios from "../api/axiosConfig";

const getAllJobs = (page = 0, size = 5) => {
    return axios.get(`/jobs?page=${page}&size=${size}`);
};

const searchJobs = (keyword, page = 0, size = 5) => {
    return axios.get(
        `/jobs/search?keyword=${keyword}&page=${page}&size=${size}`
    );
};

const createJob = (job) => {
    return axios.post("/jobs", job);
};

const getJobById = (id) => {
    return axios.get(`/jobs/${id}`);
};

const updateJob = (id, job) => {
    return axios.put(`/jobs/${id}`, job);
};

const deleteJob = (id) => {
    return axios.delete(`/jobs/${id}`);
};

export default {
    getAllJobs,
    searchJobs,
    createJob,
    getJobById,
    updateJob,
    deleteJob
};